src/webui/web_server_utils/system_monitor_mixin.py

```
@@ -135,62 +153,212 @@ def system_update_status(self) -> tuple[dict, int]:
+            RuntimeError: If the command exits non-zero or times out.
+        """
+        display_command = " ".join(command)
+        self._publish_system_update_progress(
+            phase=phase,
+            phase_index=phase_index,
+            phase_count=phase_count,
+            percent=percent_start,
+            line=f"$ {display_command}",
+        )
+
+        process = subprocess.Popen(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)